### PR TITLE
Add `agendaupdate` subcommand

### PR DESCRIPTION
**Pull request by michaelmhoffman**
Add the `agendaupdate` subcommand, the minimum viable product for #550.

Refactor `gcalcli.gcal.GoogleCalendarInterface._tsv()`, replacing the long list of if statements with a modular design using a series of `gcalcli.details.Handler` classes.

My plan is to keep making additions to the michaelmhoffman/issue-550 branch, pushing from my local machine when I have changes that implement each of the checklist items in #550. I test locally with Python 3.5-Python 3.8. Please feel free to review and apply now or when I'm done with the whole set.

### CODE REVIEW

**Comment by michaelmhoffman** [comment]
@jcrowgey This pull request fully implements #550, passes the tox tests, and is ready for review.

File: gcalcli/utils.py [hunk 1]
```diff
@@ -143,3 +143,12 @@ def agenda_time_fmt(dt, military):
     hour_min_fmt = '%H:%M' if military else '%I:%M'
     ampm = '' if military else dt.strftime('%p').lower()
     return dt.strftime(hour_min_fmt).lstrip('0') + ampm
+
+
+def is_all_day(event):
+    # XXX: currently gcalcli represents all-day events as those that both begin
```
Comment (jcrowgey) [inline gcalcli/utils.py:149]: Good point!  Is there a field in the API event structure which disambiguates these cases?

File: gcalcli/details.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+      for fieldname in handler.fieldnames)
+     for handler in HANDLERS.values())))
+
+FIELDNAMES_READONLY = frozenset(fieldname
+                                for fieldname, handler
+                                in FIELD_HANDLERS.items()
+                                if handler in HANDLERS_READONLY)
+
+_DETAILS_WITHOUT_HANDLERS = ['length', 'reminders', 'attendees',
+                             'attachments', 'end']
```
Comment (jcrowgey) [inline gcalcli/details.py:273]: Is there a way to incorporate the 'end' detail into the Time handler?

**Review by jcrowgey — APPROVED** [review]
LGTM.  Thanks for this work!

**Comment by insanum** [comment]
@michaelmhoffman would you be interested in helping us mange the gcalcli project and have commit access? Your contributions have been meaningful and I'm sure you can help guide gcalcli in the right direction.

**Comment by michaelmhoffman** [comment]
I'm happy to help when I can but I just had a new baby so probably can't
dedicate much time. Will probably keep making contributions either way.

**Comment by michaelmhoffman** [comment]
Should I use a merge commit, squash and merge, or rebase and merge?

**Comment by michaelmhoffman** [comment]
Looks like squash was what was used before.

**Pull request merged**